### PR TITLE
Zectbumo fixes 19824

### PR DESCRIPTION
**Pull request by Zectbumo**
patch for issue #19824

### CODE REVIEW

**Review by dom96 — APPROVED** [review]
LGTM. One thing I would suggest is to add more tests into https://github.com/nim-lang/Nim/blob/devel/tests/stdlib/turi.nim. Also please add an entry into https://github.com/nim-lang/Nim/blob/devel/changelog.md

**Comment by Zectbumo** [comment]
> 

done and done

File: lib/pure/uri.nim [hunk 1]
```diff
@@ -140,24 +140,26 @@ func encodeQuery*(query: openArray[(string, string)], usePlus = true,
     assert encodeQuery({: }) == ""
     assert encodeQuery({"a": "1", "b": "2"}) == "a=1&b=2"
     assert encodeQuery({"a": "1", "b": ""}) == "a=1&b"
+    assert encodeQuery({"a": "1", "b": ""}, omitEq=false, sep=';') == "a=1;b="
   for elem in query:
     # Encode the `key = value` pairs and separate them with a '&'
```
Comment (konsumlamm) [inline lib/pure/uri.nim]:
```suggestion
    # Encode the `key = value` pairs and separate them with 'sep'
```

File: lib/pure/uri.nim [hunk 2]
```diff
@@ -140,24 +140,26 @@ func encodeQuery*(query: openArray[(string, string)], usePlus = true,
     assert encodeQuery({: }) == ""
     assert encodeQuery({"a": "1", "b": "2"}) == "a=1&b=2"
     assert encodeQuery({"a": "1", "b": ""}) == "a=1&b"
+    assert encodeQuery({"a": "1", "b": ""}, omitEq=false, sep=';') == "a=1;b="
```
Comment (konsumlamm) [inline lib/pure/uri.nim]:
```suggestion
    assert encodeQuery({"a": "1", "b": ""}, omitEq = false, sep = ';') == "a=1;b="
```

File: tests/stdlib/turi.nim [hunk 3]
```diff
@@ -300,7 +302,9 @@ template main() =
 
   block: # decodeQuery
     doAssert toSeq(decodeQuery("a=1&b=0")) == @[("a", "1"), ("b", "0")]
+    doAssert toSeq(decodeQuery("a=1;b=0", sep=';')) == @[("a", "1"), ("b", "0")]
```
Comment (konsumlamm) [inline tests/stdlib/turi.nim]:
```suggestion
    doAssert toSeq(decodeQuery("a=1;b=0", sep = ';')) == @[("a", "1"), ("b", "0")]
```

File: tests/stdlib/turi.nim [hunk 4]
```diff
@@ -300,7 +302,9 @@ template main() =
...
     doAssert toSeq(decodeQuery("a=1&b=2c=6")) == @[("a", "1"), ("b", "2c=6")]
+    doAssert toSeq(decodeQuery("a=1;b=2c=6", sep=';')) == @[("a", "1"), ("b", "2c=6")]
```
Comment (konsumlamm) [inline tests/stdlib/turi.nim]:
```suggestion
    doAssert toSeq(decodeQuery("a=1;b=2c=6", sep = ';')) == @[("a", "1"), ("b", "2c=6")]
```

**Pull request merged**